app/models/theme_field.rb

```diff
@@ -4,6 +4,13 @@ class ThemeField < ActiveRecord::Base
 
   belongs_to :upload
 
+  scope :where_ordered, ->(hash) {
+    return none unless hash.present?
+
+    column, values = hash.first # order by the first element
+    where(hash).order("position(#{column}::text in '#{values.join(',')}')")
```

Comment:
this feels SQL injectiony to me, since we always order by theme_ids, why make this so generic? 

Also ... hash is always present in the callers